File: students/students/src/MON.SPA/src/models/diploma/basicDocumentTemplateSubjectModel.js
export class BasicDocumentTemplateSubjectModel {
  constructor(obj = {}) {
    this.uid = obj.uid || Math.floor((Math.random()) * 0x10000).toString(16);
    this.id = obj.id;
    this.subjectCanChange = obj.subjectCanChange;
    this.position = obj.position || 0;
    this.subjectId = obj.subjectId;
    this.subjectTypeId = obj.subjectTypeId;
    this.horarium = obj.horarium;
    this.basicDocumentSubjectId = obj.basicDocumentSubjectId;
    this.basicDocumentPartId = obj.basicDocumentPartId;
    this.templateId = obj.templateId;
    this.isHorariumHidden = obj.isHorariumHidden;
    this.modules = obj.modules || [];

    this.gradeCategory = obj.gradeCategory;
    this.subjectName = obj.subjectName;
    this.showSubjectNamePreview = obj.showSubjectNamePreview;
    this.isProfSubjectHeader = obj.isProfSubjectHeader;

  }
}

File: students/students/src/MON.Models/Diploma/Template/BasicDocumentTemplateSubjectModel.cs
namespace MON.Models.Diploma
{
    using System.Collections.Generic;

    public class BasicDocumentTemplateSubjectModel : BasicDocumentSubjectModel
    {
        public int? TemplateId { get; set; }
        public int BasicDocumentPartId { get; set; }
        public int? BasicDocumentSubjectId { get; set; }
        public string SubjectName { get; set; }
        public string SubjectNameShort { get; set; }
        public string SubjectTypeName { get; set; }
        public int? Horarium { get; set; }
        public bool IsHorariumHidden { get; set; }
        public int? BasicClassId { get; set; }
        public decimal? Grade { get; set; }
        public string GradeText { get; set; }
        public int? GradeCategory { get; set; }
        public int? SpecialNeedsGrade { get; set; }
        public int? OtherGrade { get; set; }
        public int? QualitativeGrade { get; set; }
        public decimal? NvoPoints { get; set; }
        public string FlLevel { get; set; }
        public int? FlSubjectId { get; set; }
        public string FlSubjectName { get; set; }
        public int? FlHorarium { get; set; }
        public string Ects { get; set; }
        public bool ShowFlSubject { get; set; }
        public List<BasicDocumentTemplateSubjectModel> Modules { get; set; }
        public bool ShowSubjectNamePreview { get; set; }
        /// <summary>
        /// Определя видимостта на опцията за избор на випуск в компонента за подрубрика.
        /// </summary>
        public bool IsProfSubjectHeader { get; set; }
    }

}
